### PR TITLE
SQL Extension SHOW LOG with AT

### LINKED ISSUE #4665: SQL Extension `SHOW LOG` with `AT`
The Spark SQL Extensions currently support
```
SHOW LOG (reference=identifier)? (IN catalog=identifier)?
```

It would be nice to have it support the `AT` keyword as well:
```
SHOW LOG (reference=identifier)? (AT tsOrHash=identifier)? (IN catalog=identifier)?
```

### PR DESCRIPTION
**Pull request by leewjae**
Fixes #4665

### CODE REVIEW

**Review by dimas-b — CHANGES_REQUESTED** [review]
Thanks for you contribution @leewjae !

The idea LGTM in general. Please see a couple of concerns below.

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala [hunk 1]
```diff
@@ -39,11 +40,21 @@ case class ShowLogExec(
...
   ): Seq[InternalRow] = {
+
     val refName = branch
       .map(unquoteRefName)
       .getOrElse(
         NessieUtils.getCurrentRef(api, currentCatalog, catalog).getName
       )
+
+    val ref = NessieUtils.calculateRef(refName, timestampOrHash, api)
+    NessieUtils.setCurrentRefForSpark(
```
[the remark this answers is not in the document]
Comment (leewjae) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala:51]: Now learning how the nessie code works. Changed!

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java [hunk 2]
```diff
@@ -632,6 +632,19 @@ void showLog() throws NessieConflictException, NessieNotFoundException {
         .containsExactlyElementsOf(resultList);
   }
 
+  @Test
+  void showLogAt() throws NessieConflictException, NessieNotFoundException {
+    List<SparkCommitLogEntry> resultList = createBranchCommitAndReturnLog();
+
+    // here we are skipping commit time as its variable
```
Comment (snazy) [inline integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:639]: Can you add a test using the timestamp as in `throwWhenUseShowReferencesAtTimestampWithoutTimeZone()` ?

**Review by snazy — COMMENTED** [review]
Overall LGTM - just one test case missing

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java (hunk 2 above)
Comment (leewjae) [reply integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:639]: done

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java [hunk 3]
```diff
@@ -632,6 +632,19 @@ void showLog() throws NessieConflictException, NessieNotFoundException {
...
+    assertThat(
+            sql("SHOW LOG %s AT %s IN nessie", refName, defaultHash()).stream()
```
[the remark this answers is not in the document]
Comment (leewjae) [reply integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java]: substituted with timestamp. It is expecting logs after the specified timestamp.

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala:50]: Hmmm.... this code is still calling `setCurrentRefForSpark`. Could you double check, please?

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java (hunk 2 above)
Comment (dimas-b) [reply integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:639]: The test code below uses a hash value for the `AT` clause (which is a good test). Please add another test using a timestamp with the `AT` clause.

**Comment by snazy** [comment]
@leewjae would you refer to @dimas-b comment about the test? Looks like we're otherwise good to go on this PR.

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (leewjae) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala]: Sorry, it seems that there's some issue when I commit. It has been updated. Sorry for super late reply :(

**Comment by leewjae** [comment]
> @leewjae would you refer to @dimas-b comment about the test? Looks like we're otherwise good to go on this PR.

Done.

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java (hunk 2 above)
Comment (leewjae) [reply integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:639]: done.

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java [hunk 4]
```diff
@@ -632,6 +632,45 @@ void showLog() throws NessieConflictException, NessieNotFoundException {
         .containsExactlyElementsOf(resultList);
   }
 
+  @Test
+  void showLogAt() throws NessieConflictException, NessieNotFoundException {
+    List<SparkCommitLogEntry> resultList = createBranchCommitAndReturnLog();
+
+    // here we are skipping commit time as its variable
```
[the remark this answers is not in the document]
Comment (leewjae) [reply integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:639]: Done.

**Comment by snazy** [comment]
Deferring review to @dimas-b b/c of the "requested changes". PR LGTM.

**Review by dimas-b — APPROVED** [review]
LGTM :+1: Thanks for your contribution @leewjae !

**Pull request merged**